### PR TITLE
Added Instructions for Slurm

### PR DESCRIPTION
**Pull request by gregSchwartz18**
## Why are these changes needed?
Current instructions require ssh access to the nodes.

## What do these changes do?
Provides instructions and example code for running ray on a Slurm cluster

## Related issue number
Made in response to #826

### CODE REVIEW

**Review by robertnishihara — COMMENTED** [review]
Thanks for submitting this @gregSchwartz18! This is great. Tons of people have asked about support for Slurm.

1. To what extent does this run out of the box and what changes need to be made to run it? (I'd try it but don't have a Slurm cluster on hand). E.g., `trainer.py` doesn't exist so would have to be replaced by the user.
2. I'd suggest making this a separate documentation page called "Deploying on Slurm".
3. I'd suggest including all of the code, e.g., `trainer.py` as well (just a toy example) so that people can literally copy and paste it and get it to work. If any modifications are required (hopefully not), then point those out explicitly.

File: doc/source/using-ray-on-a-cluster.rst [hunk 1]
```diff
@@ -78,3 +78,62 @@ Stopping Ray
...
+    do
+        pkill -P $(<~/pid_storage/worker${i}.pid) sleep
+    done
+
+
+start_head.sh
+
+.. code-block:: bash
+    
+    ray start --head --redis-port=6379
```
[the remark this answers is not in the document]
Comment (gregSchwartz18) [reply doc/source/using-ray-on-a-cluster.rst]: That is a much cleaner solution, thanks! I'll make the change.

**Comment by richardliaw** [comment]
Can you also put this under the proper heading for `index.rst`? This will make sure it is exposed correctly on the Table of Contents.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
Thanks for this! I'll merge this after tests pass.

**Comment by richardliaw** [comment]
@gregSchwartz18 thanks for this contribution - appreciate it!

**Comment by robertnishihara** [comment]
Thanks @gregSchwartz18!

**Comment by gregSchwartz18** [comment]
No problem. Thanks for the feedback and help pushing!

File: doc/source/using-ray-on-a-cluster.rst [hunk 2]
```diff
@@ -78,63 +78,3 @@ Stopping Ray
...
-    python trainer.py
-
-    pkill -P $(<~/pid_storage/head.pid) sleep # Closing the head and workers
-    for ((  i=1; i<=$worker_num; i++ ))
-    do
-        pkill -P $(<~/pid_storage/worker${i}.pid) sleep
-    done
-
-
-start_head.sh
```
Comment (zpymyyn) [inline doc/source/using-ray-on-a-cluster.rst]: Hi, why are those secondary scripts removed? Does the current instruction prevent slurm from killing the background ray process?

Comment (robertnishihara) [reply doc/source/using-ray-on-a-cluster.rst]: Yes, I believe so, because `--block` is passed into `ray start`.

Comment (zpymyyn) [reply doc/source/using-ray-on-a-cluster.rst]: Ok... Following the instructions, I still got [Raylet erro](https://github.com/ray-project/ray/issues/5820#issuecomment-607036313)r as follows
`IOError:[RayletClient] Connection closed 
unexpectedly. [RayletClient] Failed to push profile events.` 
I thought it was due to that slurm killed the processes.

**Comment by parthjaggi** [comment]
Having a look at the script provided on:
https://docs.ray.io/en/master/cluster/slurm.html

1. This script does not have values given for `#SBATCH --ntasks`. Is this something the user has to provide?
If I understand correctly then `--ntasks` acts as number of workers we want to create.

2. If above is true, why do we need to specify the number of nodes in the script: `#SBATCH --nodes=4`?
I am not sure, how node count should matter for running rllib. If sufficient cores are provided so that they can be broken into different workers, node count should not be relevant.

3. Also what affect will the parameter `--tasks-per-node` have on the rllib run.

**Comment by richardliaw** [comment]
@parthjaggi can you please create a new issues for these questions?

**Comment by parthjaggi** [comment]
Created at #11671.

**Pull request merged**